ucsmsdk/utils/comparesyncmo.py

```diff
@@ -144,19 +144,19 @@ def _translate_managed_object(mo, xlate_org, xlate_map):
         diff_dn = mo.dn
         if diff_dn in xlate_map:
             ref_dn = xlate_map[diff_dn]
+            _update_mo_dn_and_naming_props(mo, ref_dn)
         else:
             diff_dn = re.sub(r'[/]*[^/]+$', '', diff_dn)
-            while diff_dn is not None or diff_dn == "":
+            while diff_dn:
                 if diff_dn not in xlate_map:
```

Comment:
Its not redundant. This is for the use case where exact dn is not part of xlate_map but any parent_dn going from bottom to top present in xlate_map.
Let say ref_dn is `org-root\org-ref\ls-sp`  and diff_dn is `org-root\org-diff\ls-sp` but xlate_map is `{'org-root\org-diff': 'org-root\org-ref'}`. 
If you see a line above, diff_dn is modified to become the parent_dn and now we check if parent_dn is exists as a key in xlate_map.